### PR TITLE
Add upgrade notes entry on optional field data migration utility script

### PR DESCRIPTION
**Pull request by Kami**
Upgrade notes entry for optional migration script in https://github.com/StackStorm/st2/pull/5255.

### CODE REVIEW

File: docs/source/upgrade_notes.rst [hunk 1]
```diff
@@ -3,10 +3,69 @@
 Upgrade Notes
 =============
 
+|st2| v3.5
+----------
+
+.. _ref-upgrade-notes-v3-5:
+
+* Underlying database field type for storing large values such as action execution result has
```
Comment (cognifloyd) [inline docs/source/upgrade_notes.rst:11]:
```suggestion
* The underlying database field type for storing large values such as action execution result has
```

File: docs/source/upgrade_notes.rst [hunk 2]
```diff
@@ -3,10 +3,69 @@
...
+  v3.5 will automatically utilize this new field type.
+
+  Existing objects in the database will continue to utilize old field type.
+
+  If you want to migrate them to the new field type, you can use
+  ``st2-migrate-db-dict-field-values`` migration script which ships with |st2| v3.5. The script
+  only operates on "finalized" objects (i.e. finished executions) and it's idempotent which means
+  you can re-run it on failures or similar.
+
+  It's worth noting that running this script is optional - in most cases users primary care about
```
Comment (cognifloyd) [inline docs/source/upgrade_notes.rst:28]:
```suggestion
  It's worth noting that running this script is optional - in most cases users primarily care about
```

File: docs/source/upgrade_notes.rst [hunk 3]
```diff
@@ -3,10 +3,69 @@
...
+  performance for recent / new objects (e.g. viewing recent executions) so if you don't migrate
+  existing database field values this means retrieving those objects will still be slow, but it
+  doesn't affect newly created objects post v3.5 upgrade which will utilize new field type and
+  as such, exhibit much better performance.
+
+  By default script will run in an interactive mode and display a prompt with warning which needs
```
Comment (cognifloyd) [inline docs/source/upgrade_notes.rst:34]:
```suggestion
  By default the script will run in an interactive mode and display a prompt with a warning which needs
```

File: docs/source/upgrade_notes.rst [hunk 4]
```diff
@@ -3,10 +3,69 @@
...
+  to be acknowledged before continuing. If you want to run script in an non-interactive mode, pass
+  ``--yes`` command line argument to it.
+
+  Script also defaults to migrating data for the past 30 days. You can control time period for
+  objects which will be migrated using ``--start-dt`` and ``--end-dt`` argument as shown below.
```
Comment (cognifloyd) [inline docs/source/upgrade_notes.rst:39]:
```suggestion
  The script also defaults to migrating data for the past 30 days. You can migrate objects from
  a different time period using ``--start-dt`` and ``--end-dt`` arguments as shown below.
```

File: docs/source/upgrade_notes.rst [hunk 5]
```diff
@@ -3,10 +3,69 @@
...
+
+    # Migrate objects with creation date between April 20th, 2021 and April 25th, 2021
+    /opt/stackstorm/st2/bin/st2-migrate-db-dict-field-values --start-dt "2021-04-20T19:16:55Z" --end-dt "2021-04-25T19:26:55Z"
+
+    # Migrate object between April 20th and "now"
+    /opt/stackstorm/st2/bin/st2-migrate-db-dict-field-values --start-dt "2021-04-20T19:16:55Z" --end-dt "now"
+
+  .. note::
+
+    You are strongly recommended to create full database backup before running this script.
```
Comment (cognifloyd) [inline docs/source/upgrade_notes.rst:59]:
```suggestion
    You are strongly recommended to create a full database backup before running this script.
```

**Review by cognifloyd — APPROVED** [review]
A few minor nitpicks. But I'm fine if we merge without these changes.

File: docs/source/upgrade_notes.rst [hunk 6]
```diff
@@ -6,20 +6,75 @@ Upgrade Notes
...
+    # Migrate object between April 20th and "now"
+    /opt/stackstorm/st2/bin/st2-migrate-db-dict-field-values --start-dt "2021-04-20T19:16:55Z" --end-dt "now"
+
+  .. note::
+
+    You are strongly recommended to create a full database backup before running this script.
+
+    If you run this migration script and a need arises, you won't be able to rollback back to a
+    previous version (v3.4) because code in previous version doesn't include support for this new
+    field type (in such case you would need to restore the database backup).
```
Comment (armab) [inline docs/source/upgrade_notes.rst:72]: The upgrade note looked a bit excessive though :)

**Review by armab — APPROVED** [review]
Thanks @Kami and @cognifloyd :+1:

File: docs/source/upgrade_notes.rst (hunk 6 above)
Comment (Kami) [reply docs/source/upgrade_notes.rst:72]: Yeah, I think it's better to give more context than less.

I always hate when there is a regression / issue with a new version of some project / library and changelog / upgrade notes don't give much clue where to begin and then you need start going over the commits immediately which means more wasted time...

**Pull request merged**